cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DistributedStorageProviderNamespaceAdmin.java

```diff
@@ -55,15 +65,40 @@ public void create(NamespaceMeta namespaceMeta) throws IOException, ExploreExcep
     super.create(namespaceMeta);
     // TODO: CDAP-1519: Create base directory for filesets under namespace home
     // create HBase namespace
-    tableUtil.createNamespaceIfNotExists(getAdmin(), namespaceMeta.getNamespaceId().toId());
+    String namespace = tableUtil.getHBaseNamespace(namespaceMeta);
+    if (!Strings.isNullOrEmpty(namespaceMeta.getConfig().getHbaseNamespace())) {
+      if (!tableUtil.hasNamespace(getAdmin(), namespace)) {
+        throw new IOException(String.format("Custom mapped HBase namespace doesn't exist %s for namespace %s",
```

Comment:
Its not a part of this PR but we should include username here too. So put a TODO to do this.